spec/services/omniauth_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'rails_helper'
+
+RSpec.describe Users::Omniauth do
+  describe 'create omniauth connection' do
+    context 'existing omniauth user' do
+      let(:github_user) { FactoryBot.create(:user, :github) }
+      let(:google_user) { FactoryBot.create(:user, :google) }
+
+      it 'finds user' do
```

Comment:
Hm, interesting. I've never seen testing in a cycle and not sure it will be easy to extend them in the future. We usually use [shared examples](https://relishapp.com/rspec/rspec-core/docs/example-groups/shared-examples) in such cases. What do you think about them? Consider:
```ruby

describe '.user' do
  context 'when omniauth user exists' do
    let(:auth) { double(provider: user.provider, uid: user.uid) }
    let(:auth_service) { described_class.new(auth) }

    context 'for google' do
      let!(:user) { FactoryBot.create(:user, :google) }

      include_examples 'finds ouath user'
    end

    context 'for github' do
      let!(:user) { FactoryBot.create(:user, :github) }

      include_examples 'finds ouath user'
    end
  end
end
```
```ruby
RSpec.shared_examples 'finds ouath user' do
  it 'finds user' do
    expect(auth_service.user.uid).to eq(user.uid)
  end
end
```

Reply:
Good catch. You're right, will put shared examples here. It would fit better

P.S.: I saw a lot of testing in a cycle in feature testing, when we need to apply data driven test approach. But for service/controller testing this would be an overkill, for sure.